compiler-core/src/analyse.rs

```diff
@@ -1102,6 +1110,35 @@ impl<'a, A> ModuleAnalyzer<'a, A> {
             hydrator.disallow_new_type_variables();
             let type_ = hydrator.type_from_ast(resolved_type, environment, &mut self.problems)?;
 
+            if let Type::Named {
+                module,
+                name: type_name,
+                args,
+                ..
+            } = type_.as_ref()
+            {
+                // If the alias or type itself have any parameters, it requires a
+                // lot of work to reconstruct and correctly print the alias,
+                // so we just don't register it.
```

Comment:
What are the implications of this?

Reply:
If the user makes a non-trivial type alias like:
```gleam
type MyResult(a) = Result(a, MyError)
```
We don't print `MyResult(Int)`, for example, when a value has that type, and instead expand the alias (`Result(Int, MyError)`).
Since alias information is lost, we would have to match the `MyError` type with the error type of said value, and place the ok type into the type parameter of `MyResult`. 
This is possible, but requires storing a lot of extra information in the `Names` struct, and doing more computations when printing. So I decided to leave it out for now.